### PR TITLE
Issue 71 inventory Issue

### PR DESCRIPTION
**Pull request by RutvikS-crest**

### CODE REVIEW

File: dcnm/resource_dcnm_inventory.go [hunk 1]
```diff
@@ -538,20 +580,26 @@ func resourceDCNMInventroyUpdate(d *schema.ResourceData, m interface{}) error {
...
 				continue
 			}
 
 			switchM := extractSwitchinfo(cont)
 
 			if switchM.Selectable != "true" || switchM.Reachable != "true" {
-				log.Println("Desired switch: is not reachable or not selectable or invalid user/password or bad authentication protocol", ip)
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Warning,
+					Detail:   fmt.Sprintf("Desired switch: is not reachable or not selectable or invalid user/password or bad authentication protocol %s", ip),
```
Comment (lhercot) [inline dcnm/resource_dcnm_inventory.go:601]: probably should be "desired" based on other changes.

File: dcnm/resource_dcnm_inventory.go [hunk 2]
```diff
@@ -595,7 +646,10 @@ func resourceDCNMInventroyUpdate(d *schema.ResourceData, m interface{}) error {
 				configTimeout = configTimeout - 5
 			}
 			if migrate {
-				log.Println("Timeout occurs before going into normal mode. Hence removing it!", ip)
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Warning,
+					Detail:   fmt.Sprintf("Timeout occurs before going into normal mode. Hence removing it! %s", ip),
```
Comment (lhercot) [inline dcnm/resource_dcnm_inventory.go:651]: probably should be "timeout" based on other changes.

File: dcnm/resource_dcnm_inventory.go [hunk 3]
```diff
@@ -620,19 +677,28 @@ func resourceDCNMInventroyUpdate(d *schema.ResourceData, m interface{}) error {
...
+							Severity: diag.Warning,
+							Detail:   fmt.Sprintf("error at deletion of switch %s", err),
+						})
 					}
 				}
 			}
-			log.Println("Some switches failed to discover and deploy, resuming procedure for successfully discovered switches")
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Detail:   "Some switches failed to discover and deploy, resuming procedure for successfully discovered switches",
```
Comment (lhercot) [inline dcnm/resource_dcnm_inventory.go:689]: probably should be "some" based on other changes.

File: dcnm/resource_dcnm_inventory.go [hunk 4]
```diff
@@ -789,7 +870,7 @@ func resourceDCNMInventroyDelete(d *schema.ResourceData, m interface{}) error {
 		}
 		d.SetId(strings.Join(leftIps, ","))
 
-		return fmt.Errorf("All switches are not deleted properly")
+		return append(diags, diag.Errorf("All switches are not deleted properly")...)
```
Comment (lhercot) [inline dcnm/resource_dcnm_inventory.go:873]: probably should be "all" based on other changes.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: dcnm/resource_dcnm_inventory.go [hunk 5]
```diff
@@ -260,9 +262,11 @@ func getSwitchInfo(cont *container.Container) map[string]interface{} {
 	return sInfo
 }
 
-func resourceDCNMInventroyCreate(d *schema.ResourceData, m interface{}) error {
+func resourceDCNMInventroyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	log.Println("[DEBUG] Begining Create method ")
 
+	// Get attribute values from Terraform Config
```
[the remark this answers is not in the document]
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go [hunk 6]
```diff
@@ -303,20 +308,27 @@ func resourceDCNMInventroyCreate(d *schema.ResourceData, m interface{}) error {
...
 		}
 
 		dUrl := fmt.Sprintf("/rest/control/fabrics/%s/inventory/test-reachability", strconv.Itoa(fabricID))
 		cont, err := dcnmClient.Save(dUrl, &inv)
 		if err != nil {
-			log.Println("error at test reachability for switch : ", ip, err)
+			diags = append(diags, diag.Errorf("error at test reachability for switch %s: %s", ip, err)...)
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Detail:   fmt.Sprintf("error at test reachability for switch %s: %s", ip, err),
```
[the remark this answers is not in the document]
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Only Warning is needed at this stage

File: dcnm/resource_dcnm_inventory.go [hunk 7]
```diff
@@ -303,20 +308,27 @@ func resourceDCNMInventroyCreate(d *schema.ResourceData, m interface{}) error {
...
 			continue
 		}
 
 		switchM := extractSwitchinfo(cont)
 
 		if switchM.Selectable != "true" || switchM.Reachable != "true" {
-			log.Println("Desired switch is not reachable or not selectable or invalid user/password or bad authentication protocol", ip)
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Detail:   fmt.Sprintf("Desired switch is not reachable or not selectable or invalid user/password or bad authentication protocol %s", ip),
```
[the remark this answers is not in the document]
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go [hunk 8]
```diff
@@ -360,7 +373,10 @@ func resourceDCNMInventroyCreate(d *schema.ResourceData, m interface{}) error {
 			configTimeout = configTimeout - 5
 		}
 		if migrate {
-			log.Println("Timeout occurs before going into normal mode. Hence removing it!", ip)
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Detail:   fmt.Sprintf("Timeout occurs before going into normal mode. Hence removing it! %s", ip),
```
[the remark this answers is not in the document]
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go (hunk 1 above)
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go (hunk 2 above)
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go (hunk 3 above)
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

File: dcnm/resource_dcnm_inventory.go (hunk 4 above)
Comment (RutvikS-crest) [reply dcnm/resource_dcnm_inventory.go]: Done

**Pull request closed without merging**